### ISSUE TITLE
panic: memory wrap in unbalanced regex

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#14280]
Migrated from [rt.perl.org#123320](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=123320) (status was 'resolved')

Searchable as RT123320$

**Comment by p5pRT** [comment]
### From @mauke

#### Created by @mauke




$ perl \-e 'qr\{\(?\(DEFINE\)\}'
panic&#8203;: memory wrap at \-e line 1\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl 5.20.1:

Configured by mauke at Thu Nov  6 21:08:16 CET 2014.

Summary of my perl5 (revision 5 version 20 subversion 1) configuration:
   
  Platform:
    osname=linux, osvers=3.17.1-1-arch, archname=i686-linux
    uname='linux simplicio 3.17.1-1-arch #1 smp preempt wed oct 15 15:36:07 cest 2014 i686 gnulinux '
    config_args=''
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    use64bitint=undef, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-fwrapv -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2',
    cppflags='-fwrapv -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.9.1 20140903 (prerelease)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -fstack-protector -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib/gcc/i686-pc-linux-gnu/4.9.1/include-fixed /usr/lib /lib
    libs=-lnsl -lgdbm -ldb -ldl -lm -lcrypt -lutil -lc -lgdbm_compat
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lc
    libc=libc-2.20.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.20'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector'



@INC for perl 5.20.1:
    /home/mauke/usr/lib/perl5/site_perl/5.20.1/i686-linux
    /home/mauke/usr/lib/perl5/site_perl/5.20.1
    /home/mauke/usr/lib/perl5/5.20.1/i686-linux
    /home/mauke/usr/lib/perl5/5.20.1
    .


Environment for perl 5.20.1:
    HOME=/home/mauke
    LANG=en_US.UTF-8
    LANGUAGE (unset)
    LC_COLLATE=POSIX
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/mauke/perl5/perlbrew/bin:/home/mauke/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
    PERLBREW_BASHRC_VERSION=0.69
    PERLBREW_HOME=/home/mauke/.perlbrew
    PERLBREW_ROOT=/home/mauke/perl5/perlbrew
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @cpansprout

On Fri Nov 28 13&#8203;:48&#8203;:35 2014\, mauke\- wrote&#8203;:
> 
> This is a bug report for perl from l\.mai@&#8203;web\.de\,
> generated with the help of perlbug 1\.40 running under perl 5\.20\.1\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please describe your issue here\]
> 
> 
> $ perl \-e 'qr\{\(?\(DEFINE\)\}'
> panic&#8203;: memory wrap at \-e line 1\.

$ perl5\.21\.5  \-e 'qr\{\(?\(DEFINE\)\}'
Switch \(?\(condition\)\.\.\. not terminated in regex; marked by \<\-\- HERE in m/\(?\(DEFINE\) \<\-\- HERE / at \-e line 1\.
$ perl5\.21\.5 \-v|head \-3

This is perl 5\, version 21\, subversion 5 \(v5\.21\.5 \(v5\.21\.4\-547\-g0efa6de\)\) built for darwin\-2level

$ perl5\.21\.4  \-e 'qr\{\(?\(DEFINE\)\}'
Assertion failed&#8203;: \(\(IV\)elen >= 0\)\, function Perl\_sv\_vcatpvfn\_flags\, file sv\.c\, line 12371\.
Abort trap&#8203;: 6
$ perl5\.21\.4 \-v |head \-3

This is perl 5\, version 21\, subversion 4 \(v5\.21\.4 \(v5\.21\.3\-824\-g5cbc519\)\) built for darwin\-2level



\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'pending release'

**Comment by p5pRT** [comment]
### From @mauke

On Fri Nov 28 14&#8203;:32&#8203;:34 2014\, sprout wrote&#8203;:
> On Fri Nov 28 13&#8203;:48&#8203;:35 2014\, mauke\- wrote&#8203;:
> >
> > This is a bug report for perl from l\.mai@&#8203;web\.de\,
> > generated with the help of perlbug 1\.40 running under perl 5\.20\.1\.
> >
> >
> > \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> > \[Please describe your issue here\]
> >
> >
> > $ perl \-e 'qr\{\(?\(DEFINE\)\}'
> > panic&#8203;: memory wrap at \-e line 1\.
> 
> $ perl5\.21\.5  \-e 'qr\{\(?\(DEFINE\)\}'
> Switch \(?\(condition\)\.\.\. not terminated in regex; marked by \<\-\- HERE in
> m/\(?\(DEFINE\) \<\-\- HERE / at \-e line 1\.
> $ perl5\.21\.5 \-v|head \-3
> 
> This is perl 5\, version 21\, subversion 5 \(v5\.21\.5 \(v5\.21\.4\-547\-
> g0efa6de\)\) built for darwin\-2level
> 
> $ perl5\.21\.4  \-e 'qr\{\(?\(DEFINE\)\}'
> Assertion failed&#8203;: \(\(IV\)elen >= 0\)\, function Perl\_sv\_vcatpvfn\_flags\,
> file sv\.c\, line 12371\.
> Abort trap&#8203;: 6
> $ perl5\.21\.4 \-v |head \-3
> 
> This is perl 5\, version 21\, subversion 4 \(v5\.21\.4 \(v5\.21\.3\-824\-
> g5cbc519\)\) built for darwin\-2level

Yeah\, looks like this was fixed by commit 59685a4604d61709c23100c3754f52081ac007f1\. Sorry for the noise\.

**Comment by p5pRT** [comment]
### From @karenetheridge

I can reproduce this on darwin in all versions up to 5\.21\.4; on 5\.21\.5\, I
get&#8203;: "Switch \(?\(condition\)\.\.\. not terminated in regex; marked by \<\-\- HERE
in m/\(?\(DEFINE\) \<\-\- HERE / at \-e line 1\."

On Fri\, Nov 28\, 2014 at 1&#8203;:48 PM\, l\.mai@&#8203;web\.de \<perlbug\-followup@&#8203;perl\.org>
wrote&#8203;:

> \# New Ticket Created by  l\.mai@&#8203;web\.de
> \# Please include the string&#8203;:  \[perl \#123320\]
> \# in the subject line of all future correspondence about this issue\.
> \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=123320 >
>
>
>
> This is a bug report for perl from l\.mai@&#8203;web\.de\,
> generated with the help of perlbug 1\.40 running under perl 5\.20\.1\.
>
>
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please describe your issue here\]
>
>
> $ perl \-e 'qr\{\(?\(DEFINE\)\}'
> panic&#8203;: memory wrap at \-e line 1\.
>
>
> \[Please do not change anything below this line\]
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \-\-\-
> Flags&#8203;:
>     category=core
>     severity=low
> \-\-\-
> Site configuration information for perl 5\.20\.1&#8203;:
>
> Configured by mauke at Thu Nov  6 21&#8203;:08&#8203;:16 CET 2014\.
>
> Summary of my perl5 \(revision 5 version 20 subversion 1\) configuration&#8203;:
>
>   Platform&#8203;:
>     osname=linux\, osvers=3\.17\.1\-1\-arch\, archname=i686\-linux
>     uname='linux simplicio 3\.17\.1\-1\-arch \#1 smp preempt wed oct 15
> 15&#8203;:36&#8203;:07 cest 2014 i686 gnulinux '
>     config\_args=''
>     hint=recommended\, useposix=true\, d\_sigaction=define
>     useithreads=undef\, usemultiplicity=undef
>     use64bitint=undef\, use64bitall=undef\, uselongdouble=undef
>     usemymalloc=n\, bincompat5005=undef
>   Compiler&#8203;:
>     cc='cc'\, ccflags ='\-fwrapv \-fno\-strict\-aliasing \-pipe
> \-fstack\-protector \-I/usr/local/include \-D\_LARGEFILE\_SOURCE
> \-D\_FILE\_OFFSET\_BITS=64'\,
>     optimize='\-O2'\,
>     cppflags='\-fwrapv \-fno\-strict\-aliasing \-pipe \-fstack\-protector
> \-I/usr/local/include'
>     ccversion=''\, gccversion='4\.9\.1 20140903 \(prerelease\)'\, gccosandvers=''
>     intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
>     d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
>     ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\,
> lseeksize=8
>     alignbytes=4\, prototype=define
>   Linker and Libraries&#8203;:
>     ld='cc'\, ldflags =' \-fstack\-protector \-L/usr/local/lib'
>     libpth=/usr/local/lib
> /usr/lib/gcc/i686\-pc\-linux\-gnu/4\.9\.1/include\-fixed /usr/lib /lib
>     libs=\-lnsl \-lgdbm \-ldb \-ldl \-lm \-lcrypt \-lutil \-lc \-lgdbm\_compat
>     perllibs=\-lnsl \-ldl \-lm \-lcrypt \-lutil \-lc
>     libc=libc\-2\.20\.so\, so=so\, useshrplib=false\, libperl=libperl\.a
>     gnulibc\_version='2\.20'
>   Dynamic Linking&#8203;:
>     dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E'
>     cccdlflags='\-fPIC'\, lddlflags='\-shared \-O2 \-L/usr/local/lib
> \-fstack\-protector'
>
>
> \-\-\-
> @&#8203;INC for perl 5\.20\.1&#8203;:
>     /home/mauke/usr/lib/perl5/site\_perl/5\.20\.1/i686\-linux
>     /home/mauke/usr/lib/perl5/site\_perl/5\.20\.1
>     /home/mauke/usr/lib/perl5/5\.20\.1/i686\-linux
>     /home/mauke/usr/lib/perl5/5\.20\.1
>     \.
>
> \-\-\-
> Environment for perl 5\.20\.1&#8203;:
>     HOME=/home/mauke
>     LANG=en\_US\.UTF\-8
>     LANGUAGE \(unset\)
>     LC\_COLLATE=POSIX
>     LD\_LIBRARY\_PATH \(unset\)
>     LOGDIR \(unset\)
>
> PATH=/home/mauke/perl5/perlbrew/bin&#8203;:/home/mauke/bin&#8203;:/usr/local/sbin&#8203;:/usr/local/bin&#8203;:/usr/bin&#8203;:/usr/bin/site\_perl&#8203;:/usr/bin/vendor\_perl&#8203;:/usr/bin/core\_perl
>     PERLBREW\_BASHRC\_VERSION=0\.69
>     PERLBREW\_HOME=/home/mauke/\.perlbrew
>     PERLBREW\_ROOT=/home/mauke/perl5/perlbrew
>     PERL\_BADLANG \(unset\)
>     SHELL=/bin/bash
>
>

**Comment by p5pRT** [comment]
### From @khwilliamson

Thanks for submitting this ticket

The issue should be resolved with the release today of Perl v5\.22\, available at http&#8203;://www\.perl\.org/get\.html
If you find that the problem persists\, feel free to reopen this ticket

\-\- 
Karl Williamson for the Perl 5 porters team

**Comment by p5pRT** [comment]
@khwilliamson - Status changed from 'pending release' to 'resolved'
